examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -43,6 +52,44 @@ class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
         this._priority.on("valueChanged", () => {
             this.emit("priorityChanged");
         });
+        this.diffType = "none";
+        this.diffName = "";
+        this.diffPriority = Number.POSITIVE_INFINITY;
+    }
+    public externalNameChanged = async (savedName: string): Promise<void> => {
+        this.diffName = savedName;
+        this.diffType = "change";
+        this.emit('externalNameChanged');
+    }
+    public externalPriorityChanged = async (savedPriority: number): Promise<void> => {
+        this.diffPriority = savedPriority;
+        this.diffType = "change";
+        this.emit('externalPriorityChanged');
+    }
```

Comment:
Do these need to be Promises?

Reply:
Hah! Interestingly no! I thought they had to be promises because of the events! But I see that's not the case!